crates/build/src/manifest.rs

```diff
@@ -4,37 +4,55 @@ use std::{collections::BTreeMap, path::Path};
 
 use spin_manifest::{schema::v2, ManifestVersion};
 
+pub type DeploymentTarget = String;
+pub type DeploymentTargets = Vec<DeploymentTarget>;
+
 /// Returns a map of component IDs to [`v2::ComponentBuildConfig`]s for the
 /// given (v1 or v2) manifest path. If the manifest cannot be loaded, the
 /// function attempts fallback: if fallback succeeds, result is Ok but the load error
 /// is also returned via the second part of the return value tuple.
 pub async fn component_build_configs(
     manifest_file: impl AsRef<Path>,
-) -> Result<(Vec<ComponentBuildInfo>, Option<spin_manifest::Error>)> {
+) -> Result<(
+    Vec<ComponentBuildInfo>,
+    DeploymentTargets,
+    Result<spin_manifest::schema::v2::AppManifest, spin_manifest::Error>,
+)> {
     let manifest = spin_manifest::manifest_from_file(&manifest_file);
     match manifest {
-        Ok(manifest) => Ok((build_configs_from_manifest(manifest), None)),
-        Err(e) => fallback_load_build_configs(&manifest_file)
-            .await
-            .map(|bc| (bc, Some(e))),
+        Ok(mut manifest) => {
+            spin_manifest::normalize::normalize_manifest(&mut manifest);
+            let bc = build_configs_from_manifest(&manifest);
+            let dt = deployment_targets_from_manifest(&manifest);
+            Ok((bc, dt, Ok(manifest)))
+        }
+        Err(e) => {
+            let bc = fallback_load_build_configs(&manifest_file).await?;
```

Comment:
On reflection these should preserve (and immediately return) the original manifest load error rather than blatting it with whatever went awry during the fallback attempt.